File: db/migrate/0002_create_blogs.rb
# -*- coding: utf-8 -*-

class CreateBlogs < ActiveRecord::Migration
  def change
    create_table(:blogs) do |t|
      ## Database authenticatable
      t.string :name,               null: false
      t.string :description,        null: false
      t.string :keywords,           null: false
      t.string :url,                null: false
      t.string :image_url,          null: false
      t.string :lang,               null: false
      t.string :host,               null: false
      t.json :attrs,                null: false, default: '{}'

      t.timestamps
    end

    add_reference :blogs, :author, index: true, null: false
    add_index :blogs, :host, :unique => true
  end
end

# eof

File: priv/repo/migrations/0002_create_blogs.exs
defmodule Burp.Repo.Migrations.CreateBlogs do
  use Ecto.Migration

  def change do
    create table(:blogs) do
      add :name, :string, null: false
      add :description, :text, null: false
      add :keywords, :string, null: false
      add :url, :string, null: false
      add :image_url, :string, null: false
      add :lang, :string, null: false
      add :host, :string, null: false
      add :attrs, :map, null: false, default: %{}
      add :author_id, references(:authors, on_delete: :nothing), null: false

      timestamps()
    end

    create unique_index(:blogs, [:host])
    create index(:blogs, [:author_id])
  end
end
